realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -585,6 +622,7 @@ private void emitCopyOrUpdateMethod(JavaWriter writer) throws IOException {
                 .beginControlFlow("if (rowIndex != TableOrView.NO_MATCH)")
                     .emitStatement("realmObject = new %s()", Utils.getProxyClassName(className))
                     .emitStatement("realmObject.realm = realm")
+                    .emitStatement("realmObject.updateColumnInfo(realm.getColumnInfo(%s.class))", className)
```

Comment:
`Realm` class has following code and it requires `updateColumnInfo() on `RealmObject`.

```
    <E extends RealmObject> E get(Class<E> clazz, long rowIndex) {
        Table table = getTable(clazz);
        UncheckedRow row = table.getUncheckedRow(rowIndex);
        E result = configuration.getSchemaMediator().newInstance(clazz);
        result.row = row;
        result.realm = this;
        result.updateColumnInfo(getColumnInfo(clazz));
        return result;
    }
```

`Realm` class couldn't assign to the field of Proxy class.
If there is a way for `Realm` to assign value to Proxy class, we can remove `updateColumnInfo() from `RealmObject`.